scripts/bootloader/provision.py

```diff
@@ -151,6 +166,42 @@ def get_hashes(public_key_files, verify_hashes):
 
     return hashes
 
+def get_variable_data(psa_certification_reference):
+    # Get variable data to be written to the OTP-region.
+    #
+    # Variable data, if present, starts with variable data collection ID, followed
+    # by amount of variable data entries and the variable data entries themselves.
+    # [Collection ID][Variable count][Type][Variable data length][Variable data][Type]...
+    # 2 bytes        2 bytes         1 byte 1 byte                0-255 bytes
+    #
+    variable_data = b''
+    variable_data_count = 0
+
+    def add_variable_data(variable_data_type, data):
+        nonlocal variable_data
+        nonlocal variable_data_count
+
+        variable_data += struct.pack('B', variable_data_type)
+        variable_data += struct.pack('B', len(data))
+        variable_data += data.encode('ascii')
+        variable_data_count += 1
+
+    if psa_certification_reference:
+        if len(psa_certification_reference) > CERTIFICATION_REF_MAX_SIZE:
+            raise RuntimeError(f"PSA certification reference is too long. Maximum length is {CERTIFICATION_REF_MAX_SIZE} characters.")
+        add_variable_data(BL_VARIABLE_DATA_TYPE_PSA_CERTIFICATION_REFERENCE, psa_certification_reference)
+
+    if variable_data_count:
+        # Add the variable data header at the beginning of the variable data.
+        variable_data = struct.pack('H', BL_COLLECTION_TYPE_VARIABLE_DATA) + \
+            struct.pack('H', variable_data_count) +  \
+            variable_data
+
+        # Padding to align to 4 bytes.
+        padding_length = 4 - (len(variable_data) % 4)
```

Comment:
You seem to always add 1,2,3 or 4 bytes of padding, instead of 0 - 3.

If 4 bytes is needed, then isn't it the same as already aligned to boundary, so zero bytes of padding needed.

```suggestion
        padding_length = (4 - (len(variable_data) % 4)) % 4
```